File: Amazon_LeetCode_Python_C++/ArrayString/345_reverse_vowels_of_a_string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        v = []
        t = ['a', 'e', 'i', 'o', 'u']
        for i in range(len(s)):
            if s[i].lower() in t:
                v.append(i)
        
        s = list(s)
        l, r = 0, len(v) - 1
        while l < r:
            s[v[l]], s[v[r]] = s[v[r]], s[v[l]]
            l += 1
            r -= 1
        
        return ''.join(s)

File: Amazon_LeetCode_Python_C++/ArrayString/345_reverse_vowels_of_a_string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        vector<int> c;
        for (int i = 0; i < s.size(); i++) {
            if (tolower(s[i]) == 'a' || tolower(s[i]) == 'e' || tolower(s[i]) == 'i' || 
                tolower(s[i]) == 'o' || tolower(s[i]) == 'u') {
                c.push_back(i);
            }
        }
        int l = 0, r = c.size() - 1;
        while (l < r) {
            swap(s[c[l]], s[c[r]]);
            l++;
            r--;
        }
        return s;
    }
};
